app/src/ui/repositories-list/repository-list-item-context-menu.ts

```diff
@@ -0,0 +1,112 @@
+  repository: Repositoryish
+  shellLabel: string
+  externalEditorLabel: string | undefined
+  askForConfirmationOnRemoveRepository: boolean
+  onViewOnGitHub: (repository: Repositoryish) => void
+  onOpenInShell: (repository: Repositoryish) => void
+  onShowRepository: (repository: Repositoryish) => void
+  onOpenInExternalEditor: (repository: Repositoryish) => void
+  onRemoveRepository: (repository: Repositoryish) => void
+  onChangeRepositoryAlias: (repository: Repository) => void
+  onRemoveRepositoryAlias: (repository: Repository) => void
+}
+
+export const generateRepositoryListContextMenu = (
+  config: IRepositoryListItemContextMenuConfig
+) => {
+  const missing =
+    config.repository instanceof Repository && config.repository.missing
+  const github =
+    config.repository instanceof Repository &&
+    config.repository.gitHubRepository != null
+  const openInExternalEditor = config.externalEditorLabel
+    ? `Open in ${config.externalEditorLabel}`
+    : DefaultEditorLabel
+
+  const items: ReadonlyArray<IMenuItem> = [
+    ...buildAliasMenuItems(config),
+    {
+      label: __DARWIN__ ? 'Copy Repo Name' : 'Copy repo name',
+      action: () => clipboard.writeText(config.repository.name),
+    },
+    { type: 'separator' },
+    {
+      label: 'View on GitHub',
+      action: () => config.onViewOnGitHub(config.repository),
+      enabled: github,
+    },
+    {
+      label: `Open in ${config.shellLabel}`,
+      action: () => config.onOpenInShell(config.repository),
+      enabled: !missing,
+    },
+    {
+      label: RevealInFileManagerLabel,
+      action: () => config.onShowRepository(config.repository),
+      enabled: !missing,
+    },
+    {
+      label: openInExternalEditor,
+      action: () => config.onOpenInExternalEditor(config.repository),
+      enabled: !missing,
+    },
+    { type: 'separator' },
+    {
+      label: config.askForConfirmationOnRemoveRepository ? 'Remove…' : 'Remove',
+      action: () => config.onRemoveRepository(config.repository),
+    },
+  ]
+
+  return items
+}
+
+const buildAliasMenuItems = (
+  config: IRepositoryListItemContextMenuConfig
+): ReadonlyArray<IMenuItem> => {
+  if (
+    !(config.repository instanceof Repository) ||
+    !enableRepositoryAliases()
+  ) {
+    return []
+  }
+
+  const removeAlias = () => {
+    if (config.repository instanceof Repository) {
+      config.onRemoveRepositoryAlias(config.repository)
+    }
+  }
+
+  const changeAlias = () => {
+    if (config.repository instanceof Repository) {
+      config.onChangeRepositoryAlias(config.repository)
+    }
+  }
+
+  const verb = config.repository.alias == null ? 'Create' : 'Change'
+  const items: Array<IMenuItem> = [
+    {
+      label: __DARWIN__ ? `${verb} Alias` : `${verb} alias`,
+      action: changeAlias,
+    },
+  ]
+
+  if (config.repository.alias !== null) {
+    items.push({
+      label: __DARWIN__ ? 'Remove Alias' : 'Remove alias',
+      action: removeAlias,
+    })
+  }
+
+  return items
```

Comment:
We could simplify this a bunch by capturing repository into a const and only doing the `Repository` type check once

```suggestion
  const { repository } = config
  if (!(repository instanceof Repository) || !enableRepositoryAliases()) {
    return []
  }

  const verb = repository.alias == null ? 'Create' : 'Change'
  const items: Array<IMenuItem> = [
    {
      label: __DARWIN__ ? `${verb} Alias` : `${verb} alias`,
      action: () => config.onRemoveRepositoryAlias(repository),
    },
  ]

  if (repository.alias !== null) {
    items.push({
      label: __DARWIN__ ? 'Remove Alias' : 'Remove alias',
      action: () => config.onChangeRepositoryAlias(repository),
    })
  }

  return items
```